src/Rings/mpoly-localizations.jl

```diff
@@ -126,7 +126,23 @@ function simplify_light(S::MPolyPowersOfElement)
   end
 
   new_denom = elem_type(R)[a for a in unique!(new_denom) if !is_unit(a)]
+  fac_denom = elem_type(R)[]
 
+  # Try to factor the denominators if this is deemed possible
```

Comment:
would it be worth to make `simplify` and `simplify_light` idempotent?

Reply:
I think it tries to factor each time again? What I mean is to give them an attribute like 
`is_simplified`

Reply:
That could make sense, yes.

Reply:
I can also be delayed to a later PR. Your choice.

Reply:
Done.